mlflow/azureml/__init__.py

```diff
@@ -4,139 +4,295 @@
+    # still be accessible for import from Python 2. Therefore, we will only import from the SDK
+    # upon method invocation.
+    from azureml.core.image import ContainerImage
+    from azureml.core.model import Model as AzureModel
+
+    if run_id is not None:
+        relative_model_path = model_path
+        model_path = _get_model_log_dir(model_name=model_path, run_id=run_id)
+    else:
+        relative_model_path = model_path
+
+    model_pyfunc_conf = _load_pyfunc_conf(model_path=model_path)
+    model_python_version = model_pyfunc_conf.get(pyfunc.PY_VERSION, None)
+    if model_python_version is not None and\
+            StrictVersion(model_python_version) < StrictVersion("3.0.0"):
+        raise MlflowException(
+                message="Azure ML can only deploy models trained in Python 3 or above!",
+                error_code=INVALID_PARAMETER_VALUE)
+
+    tags = _build_tags(model_path=relative_model_path, run_id=run_id,
+                       model_python_version=model_python_version, user_tags=tags)
+
+    if image_name is None:
+        image_name = "mlflow-{uid}".format(uid=_get_azureml_resource_unique_id())
+    if model_name is None:
+        model_name = "mlflow-{uid}".format(uid=_get_azureml_resource_unique_id())
+
+    with TempDir() as tmp:
+        model_directory_path = tmp.path("model")
+        model_path = os.path.join(
+            model_directory_path,
+            _copy_file_or_tree(src=model_path, dst=model_directory_path))
+
+        registered_model = AzureModel.register(workspace=workspace, model_path=model_path,
+                                               model_name=model_name, tags=tags,
+                                               description=description)
+        eprint("Registered an Azure Model with name: `{model_name}` and version:"
+               " `{model_version}`".format(model_name=registered_model.name,
+                                           model_version=registered_model.version))
+
+        # Create an execution script (entry point) for the image's model server in the
+        # current working directory
+        execution_script_file = _create_execution_script(azure_model=registered_model)
+        # Azure ML copies the execution script into the image's application root directory by
+        # prepending "/var/azureml-app" to the specified script path. The script is then executed
+        # by referencing its path relative to the "/var/azureml-app" directory. Unfortunately,
+        # if the script path is an absolute path, Azure ML attempts to reference it directly,
+        # resulting in a failure. To circumvent this problem, we provide Azure ML with the relative
+        # script path. Because the execution script was created in the current working directory,
+        # this relative path is the script path's base name.
+        execution_script_path = os.path.basename(execution_script_file.name)
```

Comment:
Passing around typed objects in python is a bit weird -- would recommend just returning execution_script_file.name from _create_execution_script so we can expect that it's just the filename. Maybe rename it `execution_script_filename` or something.